support/infra/src/types.rs

```diff
@@ -19,26 +19,50 @@ pub struct Metadata {
     pub owner: String,
     pub location: String,
     pub size: u64,
-    pub fingerprint: Key,
+    pub fingerprint: H256,
+}
+
+impl Metadata {
+    pub fn chunk_count(&self) -> u64 {
+        let full_chunks = self.size / (FILE_CHUNK_SIZE as u64);
+        if self.size % (FILE_CHUNK_SIZE as u64) > 0 {
+            return full_chunks + 1;
+        }
+        full_chunks
+    }
+
+    pub fn chunk_ids(&self) -> impl Iterator<Item = ChunkId> {
+        (0..self.chunk_count()).map(|n| ChunkId(n))
+    }
+}
+
+/// Typed u64 representing the index of a file [`Chunk`]. Indexed from 0.
+#[derive(Clone)]
+pub struct ChunkId(pub u64);
```

Comment:
If not really necessary, I'd rather keep the inner type private and create a method to access it like `pub fn inner(&self) -> u64` in the `ChunkId` impl.

Reply:
Or maybe we can just use it as a `newtype` with `type ChunkId = u64` since the only method in this Impl simply calls `to_be_bytes()`